### DISCUSSION TITLE
CUE encoding

### DISCUSSION THREAD

**Discussion by roman-mazur** [cue-lang/cue#1952] [Ideas] [answered]
I found a case recently where I'd like to define a command that writes or prints CUE code. Would it make sense to introduce `encoding/cue` package with a function `Marshal` just like in `encoding/json` or `encoding/yaml`.

To illustrate
```
import (
  "encoding/cue"

  "tool/cli"
)

output: foo: "bar"

command: print: cli.Print & { text: cue.Marshal(output) }
```

I understand that I can solve some of the cases with `cue eval -e output`, but it still seems more limited compared to having the CUE encoding.

**Comment by myitcv** [comment, marked as answer]
Please see https://github.com/cue-lang/cue/issues/1852
